### PR TITLE
RA-445

### PR DESCRIPTION
**Pull request by aoller**

### CODE REVIEW

File: VAS.DB/CouchbaseStorage.cs [hunk 1]
```diff
@@ -218,6 +218,43 @@ public void Fill (IStorable storable)
 			}
 		}
 
+		public void StoreAll<T> (IEnumerable<T> tt, bool forceUpdate = false) where T : IStorable
```
Comment (ylatuya) [inline VAS.DB/CouchbaseStorage.cs]: This function is a copy paste of the Store function.
This function needs to be added to the IStorage API.
I would not change the name of the function and keep using Store<T> (IEnumerable<T> items, bool forceUpdate = false)
Use a meaningful name for the vairable tt.

Comment (ylatuya) [inline VAS.DB/CouchbaseStorage.cs]: The new function is missing unit tests

File: VAS.Core/Serialization/Serializer.cs [hunk 2]
```diff
@@ -162,7 +162,7 @@ public T LoadSafe<T> (string filepath)
 				settings.ObjectCreationHandling = ObjectCreationHandling.Replace;
 				settings.Converters.Add (new VersionConverter ());
 				settings.Converters.Add (new VASConverter (true));
-				//settings.ReferenceResolver = new IdReferenceResolver ();
+				settings.ReferenceLoopHandling = ReferenceLoopHandling.Serialize;
```
Comment (ylatuya) [inline VAS.Core/Serialization/Serializer.cs]: Why is this needed? Which new object introduces loop references?

Comment (aoller) [inline VAS.Core/Serialization/Serializer.cs]: SummonerImage serialization requires this flag. In fact I didn't understant why it's needed. All uint tests works fine with this flag set.

File: VAS.DB/CouchbaseStorage.cs (hunk 1 above)
Comment (aoller) [inline VAS.DB/CouchbaseStorage.cs]: Changed

File: VAS.Core/Serialization/Serializer.cs (hunk 2 above)
Comment (ylatuya) [inline VAS.Core/Serialization/Serializer.cs]: There are no loop references in this Storable, there is probably probably something wrong somewhere else because it shouldn't be needed at all.

File: VAS.DB/CouchbaseStorage.cs [hunk 3]
```diff
@@ -218,6 +218,43 @@ public void Fill (IStorable storable)
 			}
 		}
 
+		public void Store<T> (IEnumerable<T> storableEnumerable, bool forceUpdate = false) where T : IStorable
```
Comment (ylatuya) [inline VAS.DB/CouchbaseStorage.cs]: This function is a c&p of the Store one with one element

File: VAS.Tests/DB/TestStorage.cs [hunk 4]
```diff
@@ -467,5 +467,31 @@ public void TestPlaylistDelete ()
...
+			Assert.IsNotNull (p1o);
+			Assert.AreEqual (p1.ID, p1o.ID);
+			Assert.AreEqual (p1.Name, p1o.Name);
+			Assert.IsNotNull (p2o);
+			Assert.AreEqual (p2.ID, p2o.ID);
+			Assert.AreEqual (p2.Name, p2o.Name);
+		}
+	}
+
+	class PlayerTest : Player
```
Comment (ylatuya) [inline VAS.Tests/DB/TestStorage.cs]: There is already a PlayerDummy

File: VAS.DB/CouchbaseStorage.cs (hunk 3 above)
Comment (aoller) [inline VAS.DB/CouchbaseStorage.cs]: Refactored

File: VAS.Tests/DB/TestStorage.cs (hunk 4 above)
Comment (aoller) [inline VAS.Tests/DB/TestStorage.cs]: Changed

File: VAS.Tests/DB/TestStorage.cs [hunk 5]
```diff
@@ -467,5 +468,27 @@ public void TestPlaylistDelete ()
 			// Assert
 			Assert.AreEqual (1, db.DocumentCount);
 		}
+
+		[Test ()]
+		public void TestStorableEnumerable ()
+		{
+			// Arrange
+			PlayerDummy p1 = new Utils.PlayerDummy { Name = "P1" };
```
Comment (ylatuya) [inline VAS.Tests/DB/TestStorage.cs]: Using this the PlayerDummy defined in the using?

Comment (aoller) [inline VAS.Tests/DB/TestStorage.cs]: Changed
